Resources/js/translator.js

```diff
@@ -401,202 +597,10 @@ var Translator = (function(document, undefined) {
-                id,
-                domain,
-                locale,
-                this.locale,
-                this.fallback
-            );
-
-            return replace_placeholders(_message, parameters || {});
-        },
-
-        /**
-         * Translates the given choice message by choosing a translation according to a number.
-         *
-         * @param {String} id               The message id
-         * @param {Number} number           The number to use to find the indice of the message
-         * @param {Object} [parameters]     An array of parameters for the message
-         * @param {String} [domain]         The domain for the message or null to guess it
-         * @param {String} [locale]         The locale or null to use the default
-         * @return {String}                 The translated string
-         * @api public
-         */
-        transChoice: function(id, number, parameters, domain, locale) {
-            var _message = get_message(
-                id,
-                domain,
-                locale,
-                this.locale,
-                this.fallback
-            );
-
-            var _number  = parseInt(number, 10);
-
-            if (undefined != _message && !isNaN(_number)) {
-                _message = pluralize(
-                    _message,
-                    _number,
-                    locale || this.locale || this.fallback
-                );
-            }
-
-            return replace_placeholders(_message, parameters || {});
-        },
-
-        /**
-         * Loads translations from JSON.
-         *
-         * @param {String} data     A JSON string or object literal
-         * @return {Object}         Translator
-         * @api public
-         */
-        fromJSON: function(data) {
-            if (typeof data === 'string') {
-                data = JSON.parse(data);
-            }
-
-            if (data.locale) {
-                this.locale = data.locale;
-            }
-
-            if (data.fallback) {
-                this.fallback = data.fallback;
-            }
-
-            if (data.defaultDomain) {
-                this.defaultDomain = data.defaultDomain;
-            }
-
-            if (data.translations) {
-                for (var locale in data.translations) {
-                    for (var domain in data.translations[locale]) {
-                        for (var id in data.translations[locale][domain]) {
-                            this.add(id, data.translations[locale][domain][id], domain, locale);
-                        }
-                    }
-                }
-            }
-
-            return this;
-        },
-
-        /**
-         * @api public
-         */
-        reset: function() {
-            _messages   = {};
-            _domains    = [];
-            this.locale = get_current_locale();
+        if (typeof document !== 'undefined') {
+            return document.documentElement.lang.replace('-', '_');
         }
```

Comment:
you should return ``en`` as default locale (or use a different detection strategy). Putting it as ``null`` is a bad idea.